ngx-fudis/projects/ngx-fudis/src/lib/components/section/section.component.ts

```diff
@@ -129,25 +137,31 @@ export class SectionComponent extends TooltipApiDirective implements OnInit, OnC
     this._setSectionId();
 
     this._headingId = `${this.id}-heading`;
-
     this._classList = this._getClasses();
     this._title = this.title;
     this._addToErrorSummary();
   }
 
-  ngOnChanges(): void {
-    this._classList = this._getClasses();
+  ngOnChanges(changes: FudisComponentChanges<SectionComponent>): void {
+    if (changes.classes?.currentValue) {
+      this._classList = this._getClasses();
+    }
 
-    if (this.title !== this._title && this.id) {
-      this._title = this.title;
-      this._addToErrorSummary();
+    if (changes.title?.currentValue) {
+      if (this.title !== this._title && this.id) {
+        this._title = this.title;
+        this._addToErrorSummary();
+      }
     }
```

Comment:
These changes are a good addition, but to make them even better, the comparison should be changes.x.currentValue !== changes.x.previousValue
(this pattern should be fixed to other places as well we are using FudisComponentChanges checks)